pkg/workflow/providers/multicluster/deploy.go

```diff
@@ -0,0 +1,210 @@
+}
+
+// Deploy execute deploy workflow step
+func (executor *deployWorkflowStepExecutor) Deploy(ctx context.Context, policyNames []string, parallelism int) (bool, string, error) {
+	policies, err := selectPolicies(executor.af.Policies, policyNames)
+	if err != nil {
+		return false, "", err
+	}
+	components, err := loadComponents(ctx, executor.cli, executor.af, executor.af.Components)
+	if err != nil {
+		return false, "", err
+	}
+	placements, err := pkgpolicy.GetPlacementsFromTopologyPolicies(ctx, executor.cli, executor.af.Namespace, policies, resourcekeeper.AllowCrossNamespaceResource)
+	if err != nil {
+		return false, "", err
+	}
+	components, err = overrideConfiguration(policies, components)
+	if err != nil {
+		return false, "", err
+	}
+	return applyComponents(executor.apply, executor.healthCheck, components, placements, parallelism)
+}
+
+func selectPolicies(policies []v1beta1.AppPolicy, policyNames []string) ([]v1beta1.AppPolicy, error) {
+	policyMap := make(map[string]v1beta1.AppPolicy)
+	for _, policy := range policies {
+		policyMap[policy.Name] = policy
+	}
+	var selectedPolicies []v1beta1.AppPolicy
+	for _, policyName := range policyNames {
+		if policy, found := policyMap[policyName]; found {
+			selectedPolicies = append(selectedPolicies, policy)
+		} else {
+			return nil, errors.Errorf("policy %s not found", policyName)
+		}
+	}
+	return selectedPolicies, nil
+}
+
+func loadComponents(ctx context.Context, cli client.Client, af *appfile.Appfile, components []common.ApplicationComponent) ([]common.ApplicationComponent, error) {
+	var loadedComponents []common.ApplicationComponent
+	for _, comp := range components {
+		loadedComp, err := af.LoadDynamicComponent(ctx, cli, comp.DeepCopy())
+		if err != nil {
+			return nil, err
+		}
+		loadedComponents = append(loadedComponents, *loadedComp)
+	}
+	return loadedComponents, nil
+}
+
+func overrideConfiguration(policies []v1beta1.AppPolicy, components []common.ApplicationComponent) ([]common.ApplicationComponent, error) {
+	var err error
+	for _, policy := range policies {
+		if policy.Type == v1alpha1.OverridePolicyType {
+			overrideSpec := &v1alpha1.OverridePolicySpec{}
+			if err := utils.StrictUnmarshal(policy.Properties.Raw, overrideSpec); err != nil {
+				return nil, errors.Wrapf(err, "failed to parse override policy %s", policy.Name)
+			}
+			components, err = envbinding.PatchComponents(components, overrideSpec.Components, overrideSpec.Selector)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to apply override policy %s", policy.Name)
+			}
+		}
+	}
+	return components, nil
+}
+
+type applyTask struct {
+	component common.ApplicationComponent
+	placement v1alpha1.PlacementDecision
+}
+
+func (t *applyTask) key() string {
+	return fmt.Sprintf("%s/%s/%s", t.placement.Cluster, t.placement.Namespace, t.component.Name)
+}
+
+func (t *applyTask) dependents() []string {
+	var dependents []string
+	for _, dependent := range t.component.DependsOn {
```

Comment:
Are we gonna add `inputs/outputs` as dependents in this PR?

Reply:
We will move on to the Inputs/Outputs in the future.